bolt/lib/Rewrite/RewriteInstance.cpp

```diff
@@ -4788,13 +4788,25 @@ void RewriteInstance::updateELFSymbolTable(
     if (!IsDynSym && shouldStrip(Symbol))
       continue;
 
+    Expected<StringRef> SymbolName = Symbol.getName(StringSection);
```

Comment:
Can we move the code that checks for special symbols here and skip function matching for them all together? Otherwise we are checking for the markers in several locations unnecessarily.

Reply:
There are non-trivial conditions (https://github.com/llvm/llvm-project/pull/92713/files#diff-4be6195ccdb53447f870b9b26a36d0b916ace3cbd6efe595bb3cbffd53cf0dacL4902) that guard updating those special symbols. NFC testing shows it may not be safe to just move these checks.